components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -9,15 +9,103 @@
 #include <utility>
 
 #include "base/bind.h"
+#include "base/task/post_task.h"
+#include "brave/components/brave_shields/common/brave_shield_constants.h"
 #include "components/content_settings/core/browser/content_settings_pref.h"
 #include "components/content_settings/core/browser/website_settings_registry.h"
+#include "components/content_settings/core/common/content_settings_utils.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
 
 namespace content_settings {
 
+namespace {
+
+Rule CloneRule(const Rule& rule, bool reverse_patterns = false) {
+  auto secondary_pattern = rule.secondary_pattern;
+  if (secondary_pattern.Compare(
+          // TODO(bridiver) convert this to a brave_shields constant
+          ContentSettingsPattern::FromString("https://firstParty/*")) ==
+      ContentSettingsPattern::IDENTITY) {
```

Comment:
do we really need to compare like this? Why not just compare with `==`?

Reply:
in this case we can, I think this changed from a different comparison and just left the `Compare`